jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -136,8 +136,151 @@ ecma_typedarray_create_object_with_typedarray (ecma_object_t *typedarray_p, /**<
+        case LIT_MAGIC_STRING_INT8_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (int8_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT8_ARRAY_UL:
+        case LIT_MAGIC_STRING_UINT8_CLAMPED_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (uint8_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_INT16_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (int16_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT16_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (uint16_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_INT32_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (int32_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT32_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (uint32_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_FLOAT32_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (float);
+          break;
+        }
+        case LIT_MAGIC_STRING_FLOAT64_ARRAY_UL:
+        {
+          GET_VALUE_FROM_ARRAYBUFFER (double);
+          break;
+        }
+        default:
+        {
+          JERRY_UNREACHABLE ();
+        }
+      }
+      /** check dst type and set value to dst typedarray */
+      switch (class_id)
+      {
+        case LIT_MAGIC_STRING_INT8_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (int8_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT8_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (uint8_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT8_CLAMPED_ARRAY_UL:
+        {
+          uint8_t value;
+          int64_t tmp_int;
+
+          if (tmp >= 255)
+          {
+            value = 255;
+          }
+          else if (tmp <= 0)
+          {
+            value = 0;
+          }
+          else
+          {
+            tmp_int = (int64_t) tmp;
+            value = (uint8_t) tmp_int;
+
+            if (value + 0.5 < tmp
+                || (value + 0.5 == tmp && value % 2 == 1))
+            {
+              value ++;
+            }
+          }
+          *((uint8_t *) dst_buf) = value;
+          break;
+        }
+        case LIT_MAGIC_STRING_INT16_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (int16_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT16_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (uint16_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_INT32_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (int32_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_UINT32_ARRAY_UL:
+        {
+          SET_VALUE_TO_ARRAYBUFFER (uint32_t);
+          break;
+        }
+        case LIT_MAGIC_STRING_FLOAT32_ARRAY_UL:
+        {
+          *((float *) dst_buf) = (float) tmp;
+          break;
+        }
+        case LIT_MAGIC_STRING_FLOAT64_ARRAY_UL:
+        {
+          *((double *) dst_buf) = (double) tmp;
```

Comment:
Actually double support is not required for jerry. Perhaps we should make it optional.